src/main/java/de/tum/in/www1/artemis/service/exam/ExamAccessService.java

```diff
@@ -84,17 +84,21 @@ public StudentExam getExamInCourseElseThrow(Long courseId, Long examId) {
             studentExam = optionalStudentExam.get();
         }
         else {
-            // Only Test Exams can be self-created by the user.
             Exam examWithExerciseGroupsAndExercises = examRepository.findWithExerciseGroupsAndExercisesByIdOrElseThrow(examId);
+            // An exam can be started 5 minutes before the start time, which is when programming exercises are unlocked
+            boolean canExamBeStarted = ZonedDateTime.now().isAfter(ExamDateService.getExamProgrammingExerciseUnlockDate(examWithExerciseGroupsAndExercises));
```

Comment:
Should we also check if the exam is not yet over? Otherwise we might get weird cases where a student can create an exam after it is finished

Reply:
done